code/script.js

```diff
@@ -1,10 +1,270 @@
 // All the DOM selectors stored as short variables
-const chat = document.getElementById('chat')
+const chat = document.getElementById('chat');
+const form = document.getElementById('name-form');
+const inputWrapper = document.getElementById('input-wrapper');
+const nameInput = document.getElementById('name-input');
 
 // Global variables, if you need any, declared here
 
+let answer = 1;
+
 // Functions declared here
 
+const handleNameInput = (event) => {
+  event.preventDefault()
+  const name = nameInput.value;
+  showMessage(name, 'user');
+  nameInput.value = "";
+  inputWrapper.innerHTML = "";
+  setTimeout(() => showMessage(`Hi ${name}! I am here to assist you booking a trip.`, 'bot'),1000);
+  setTimeout(() => askWhereTo(),2500);
+}
+
+const askWhereTo = () => {
+  showMessage("Are you planning a trip within Europe or outside Europe?","bot");
+
+  inputWrapper.innerHTML=`
+  <button id="button-europe">
+    Within Europe
+  </button>
+  <button id="button-outside">
+    Outside Europe
+  </button>
+  `
+  document
+  .getElementById("button-europe")
+  .addEventListener("click", () => {
+  showMessage("Within Europe", "user")
+  inputWrapper.innerHTML = "";
+  setTimeout(() => showMessage("Wonderful! There is so much to explore in Europe.", "bot"), 1500);
+  setTimeout(() => showMessage("We have a great variety of destinations at the moment.", "bot"),3000);
+  setTimeout(() => askType(), 5000);
+  })
+
+  document
+  .getElementById("button-outside")
+  .addEventListener("click", () => {
+  showMessage("Outside Europe", "user")
+  inputWrapper.innerHTML = "";
+  setTimeout(() => showMessage("Great, so somewhere far then!", "bot"),1500);
+  setTimeout(() => showMessage("We have a great variety of destinations at the moment.", "bot"),3000);
+  setTimeout(() => askType(), 5000);
+  })
+}
+
+
+const askType = () => {
+ showMessage("What kind of trip are you planning?", "bot");
+
+    inputWrapper.innerHTML=`
+  <button id="button-solotrip">
+    Solo trip
+  </button>
+  <button id="button-friendstrip">
+  Friends trip
+  </button>
+  <button id="button-romantictrip">
+    Romantic trip
+  </button>
+  <button id="button-familytrip">
+  Family trip
+  </button>
+  `
+
+  chat.scrollTop = chat.scrollHeight;
+  document
+  .getElementById("button-solotrip")
+  .addEventListener("click", () => {
+  showMessage("A solo trip", "user")
+  answer = 1;
```

Comment:
you could create this variable here and pass it as a parameter to the next function 🤓